R/pkg/R/sparkR.R

```diff
@@ -373,8 +373,13 @@ sparkR.session <- function(
     overrideEnvs(sparkConfigMap, paramMap)
   }
 
+  deployMode <- ""
+  if (exists("spark.submit.deployMode", envir = sparkConfigMap)) {
+    deployMode <- sparkConfigMap[["spark.submit.deployMode"]]
+  }
```

Comment:
Yeah, but I'd like to keep consistent convention with ```sparkHome``` and ```master``` which use "" rather than NULL as default value. Then we don't need to deal with NULL in ```sparkCheckInstall```.